src/FFMpeg/Media/Video.php

```diff
@@ -129,6 +129,56 @@ public function save(FormatInterface $format, $outputPathfile)
             }
         }
 
+        // Merge Filters into one command
+        $videoFilterVars = $videoFilterProcesses = [];
+        for($i=0;$i<count($commands);$i++) {
+            $command = $commands[$i];
+            if ( $command == '-vf' ) {
+                $commandSplits = explode(";", $commands[$i + 1]);
+                if ( count($commandSplits) == 1 ) {
+                    $commandSplit = $commandSplits[0];
+                    $command = trim($commandSplit);
+                    if ( preg_match("/^\[in\](.*?)\[out\]$/is", $command, $match) ) {
+                        $videoFilterProcesses[] = $match[1];
+                    } else {
+                        $videoFilterProcesses[] = $command;   
+                    }
+                } else {
+                    foreach($commandSplits as $commandSplit) {
+                        $command = trim($commandSplit);
+                        if ( preg_match("/^\[[^\]]+\](.*?)\[[^\]]+\]$/is", $command, $match) ) {
+                            $videoFilterProcesses[] = $match[1];
+                        } else {
+                            $videoFilterVars[] = $command;
+                        }                
+                    }
+                }
+                unset($commands[$i]);
+                unset($commands[$i + 1]);
+                $i++;
+            }
+        }
+        $videoFilterCommands = $videoFilterVars;
+        $lastInput = 'in';
+        foreach($videoFilterProcesses as $i => $process) {
+            $command = '[' . $lastInput .']';
+            $command .= $process;
+            $lastInput = 'p' . $i;
+            if ( $i == count($videoFilterProcesses) - 1 ) {
+                $command .= '[out]';
+            } else {
+                $command .= '[' . $lastInput . ']';
+            }
+            
+            $videoFilterCommands[] = $command;
+        }
+        $videoFilterCommand = implode(";", $videoFilterCommands);
+        
+        if ( $videoFilterCommand ) {
+            $commands[] = '-vf';
+            $commands[] = $videoFilterCommand;
+        }
```

Comment:
the commands generated by the filter is not clean enough to be used by the `save` function. we need to do the ordering, `vf` is the procedure to do the ordering filter. you are hard to generate only one perfer command as only one `vf` command is allowed but you have mutiple filter classes generating multiple `vf` command. so now we have no choice, one is to modify all classes to generate a special return for `vf`, and one is to merge multiple `vf` into one in the preprocess. i think the second way is the best for our situation.